osu.Desktop/Updater/AppImageUpdateManager.cs

```diff
@@ -0,0 +1,564 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.Versioning;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game;
+using osu.Game.Graphics;
+using osu.Game.Online.API;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+using osu.Game.Updater;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Desktop.Updater
+{
+    /// <summary>
+    /// An update manager that shows notifications if a newer release is detected.<para />
+    /// Updates the AppImage and backups the previous version with a .zs-old suffix.
+    /// </summary>
+    [SupportedOSPlatform("linux")]
+    public class AppImageUpdateManager : UpdateManager
+    {
+        /// <summary>
+        /// Implements appimageupdatetool functionality via cli
+        /// </summary>
+        private class AppImageUpdateTool
```

Comment:
Move this to a different file as a `partial class` of this file.

Reply:
I don't know about partials. This should probably live as a separate top-levelc lass in `osu.Desktop/Linux` (which doesn't currently exist yet).